src/librustc_trans/trans/mir/rvalue.rs

```diff
@@ -44,7 +44,15 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
         match *rvalue {
             mir::Rvalue::Use(ref operand) => {
-                self.trans_operand_into(bcx, dest.llval, operand);
+                // FIXME: consider not copying constants through stack. (fixable by translating
+                // constants into OperandValue::Ref, why don’t we do that yet if we don’t?)
+                let operand = self.trans_operand(bcx, operand);
+                self.store_operand(bcx, dest.llval, operand);
+                match operand.val {
+                    OperandValue::Ref(v) => self.drop_fill(bcx, v, operand.ty),
+                    OperandValue::Immediate(_) |
```

Comment:
I’m not sure about `Immediate` not needing any treatment here. `Immediate` only happens if `common::type_is_immediate` returns true, but to me it is not obvious that the function will never produce `true` in case of type having a `Drop` somewhere in it.